core/state_processor.go

```diff
@@ -126,9 +126,10 @@ func applyTransaction(msg types.Message, config *params.ChainConfig, bc ChainCon
 	}
 
 	// Set the receipt logs and create the bloom filter.
-	receipt.Logs = statedb.GetLogs(tx.Hash())
+	blockhash := header.Hash()
```

Comment:
I'm still not super happy about adding a `header.hash()` here. On a full-sync, doing 1 billion transactions, we do 1 billion extra hashes. Perhaps it's not worth bothering about, but we _could_ change the signature to take the `header.Number` and the `hash` as input to `applyTransaction`, insead of passing the full `*types.Header`.

Reply:
Alright, I changed it so that we don't use the header hash anymore